tests/baselines/reference/propertyAccessChain.types

```diff
@@ -89,3 +89,12 @@ o1?.b ? 1 : 0;
 >1 : 1
 >0 : 0
 
+// GH#36031
+o2?.b!.c
+>o2?.b!.c : string
+>o2?.b! : { c: string; }

```

Comment:
IMO this should also include `undefined` as the assertion only asserts that `b` is defined, but not `o2`